### PR TITLE
Updates and expands Kafka support

### PR DESCRIPTION
**Pull request by JustinMoss**

### CODE REVIEW

**Comment by keithelder** [comment]
@JustinMoss what's the timeline on this being fully baked? Very excited to see this!

**Comment by JustinMoss** [comment]
@keithelder Assuming no extra feedback, it's about ready for v1, but we still two other features we want to add in. Transactional in Kafka does not easily follow the pattern in RockLib.Messaging so it's taking more work to figure out. We also have a "Replay" feature that we haven't looked into yet.

File: docs/Kafka.md [hunk 1]
```diff
@@ -17,6 +17,45 @@ The KafkaSender class can be directly instantiated and has the following paramet
...
+
+To add an KafkaSender to a service collection for dependency injection, use the `AddKafkaSender` method, optionally passing in a `configureOptions` callback:
+
+```c#
+services.AddKafkaSender("MySender", options =>
+{
+    options.Topic = "test";
+    options.BootstrapServers = "localhost:9092";
+    options.MessageTimeoutMs = 10000;
+    options.Config = new ProducerConfig() { SocketNagleDisable = true };
```
Comment (bfriesen) [inline docs/Kafka.md:30]:
```suggestion
```

File: docs/Kafka.md [hunk 2]
```diff
@@ -17,6 +17,45 @@ The KafkaSender class can be directly instantiated and has the following paramet
...
+
+/* appsettings.json:
+{
+    "MyKafkaSender": {
+        "Topic": "test",
+        "BootstrapServers": "localhost:9092",
+        "MessageTimeoutMs": 10000,
+        "Config": {
+            "SocketNagleDisable": true
+        }
```
Comment (bfriesen) [inline docs/Kafka.md:51]:
```suggestion
        "MessageTimeoutMs": 10000
```

File: docs/Kafka.md [hunk 3]
```diff
@@ -72,6 +111,45 @@ The KafkaReceiver class can be directly instantiated and has the following param
...
+
+To add an KafkaReceiver to a service collection for dependency injection, use the `AddKafkaReceiver` method, optionally passing in a `configureOptions` callback:
+
+```c#
+services.AddKafkaReceiver("MyReceiver", options =>
+{
+    options.Topic = "test";
+    options.BootstrapServers = "localhost:9092";
+    options.GroupId = "test-consumer-group";
+    options.Config = new ConsumerConfig() { SocketNagleDisable = true };
```
Comment (bfriesen) [inline docs/Kafka.md:124]:
```suggestion
```

File: docs/Kafka.md [hunk 4]
```diff
@@ -72,6 +111,45 @@ The KafkaReceiver class can be directly instantiated and has the following param
...
+
+/* appsettings.json:
+{
+    "MyKafkaReceiver": {
+        "Topic": "test",
+        "BootstrapServers": "localhost:9092",
+        "GroupId": "test-consumer-group",
+        "Config": {
+            "SocketNagleDisable": true
+        }
```
Comment (bfriesen) [inline docs/Kafka.md:145]:
```suggestion
        "GroupId": "test-consumer-group"
```

**Comment by JustinMoss** [comment]
Yep, thought of this after I left. Thanks!

**Pull request merged**